inject/src/main/java/io/micronaut/context/DefaultBeanContext.java

```diff
@@ -1770,13 +1770,70 @@ protected Iterable<BeanConfiguration> resolveBeanConfigurations() {
      * Initialize the event listeners.
      */
     protected void initializeEventListeners() {
-        final Map<Class<?>, List<BeanCreatedEventListener>> beanCreatedListeners = loadListeners(BeanCreatedEventListener.class);
+        final Map<Class<?>, List<BeanCreatedEventListener<?>>> beanCreatedListeners = loadCreatedListeners();
         beanCreatedListeners.put(AnnotationProcessor.class, Collections.singletonList(new AnnotationProcessorListener()));
         final Map<Class<?>, List<BeanInitializedEventListener>> beanInitializedListeners = loadListeners(BeanInitializedEventListener.class);
         this.beanCreationEventListeners = beanCreatedListeners.entrySet();
         this.beanInitializedEventListeners = beanInitializedListeners.entrySet();
     }
 
+    private void handleEagerInitializedDependencies(BeanDefinition<?> listener,
+                                                    Argument<?> listensTo,
+                                                    List<List<Argument<?>>> targets) {
+        if (LOG.isWarnEnabled()) {
+            List<String> paths = new ArrayList<>(targets.size());
+            for (List<Argument<?>> line: targets) {
+                paths.add("    " + line.stream()
+                        .map(Argument::getType)
+                        .map(Class::getName)
+                        .collect(Collectors.joining(AbstractBeanResolutionContext.DefaultPath.RIGHT_ARROW)));
+            }
+            LOG.warn("The bean created event listener {} will not be executed because one or more other bean created event listeners inject {}:\n" +
+                    "{}\n" +
+                    "Change at least one point in the path to be lazy initialized by injecting a provider to avoid this issue", listener.getBeanType().getName(), listensTo.getType().getName(), String.join("\n", paths));
+        }
+    }
+
+    @NonNull
+    private Map<Class<?>, List<BeanCreatedEventListener<?>>> loadCreatedListeners() {
+        final Collection<BeanDefinition<BeanCreatedEventListener>> beanDefinitions = getBeanDefinitions(BeanCreatedEventListener.class);
```

Comment:
Is it possible there are no listeners? In that case, I wouldn't create all the collections. Personally, I would move some of the collection initialization just before the block they are going to be used, not having everything at the top.

Reply:
Would changing it to:

```
    private Map<Class<?>, List<BeanCreatedEventListener<?>>> loadCreatedListeners() {
        final Collection<BeanDefinition<BeanCreatedEventListener>> beanDefinitions = getBeanDefinitions(BeanCreatedEventListener.class);
        if (beanDefinitions.isEmpty()) {
            return Collections.emptyMap();
        }
```

suffice?

Reply:
yeah

Reply:
Done in https://github.com/micronaut-projects/micronaut-core/pull/7222/commits/20fcf654316b029fc81d4a6f933732817df1e775